presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -73,6 +73,13 @@
     Page getOutput();
 
     /**
+     * After calling this method operator should revoke all reserved revocable memory
+     */
+    default void revokeMemory()
```

Comment:
what will happen when operator `requestMemory` future is still pending? Is there a way to cancel such requests. If not, I think there might be a deadlock. Think about it.